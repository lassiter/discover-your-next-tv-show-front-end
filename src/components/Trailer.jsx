import React, { Component } from 'react'
import ModalVideo from 'react-modal-video'
import Axios from 'axios';
import styled from 'styled-components'
import { isNotNullOrUndefined } from '../helpers/typeChecker'

const Button = styled.button`
  border-color: black;
  color: white;
  background: black;
`

export default class Trailer extends Component {
  constructor () {
    super()
    this.state = {
      loading: true,
      isOpen: false,
      youTubeKey: null
    }
    this.openModal = () => {
      this.setState({isOpen: true})
    }
  }

  componentDidMount(){
    Axios.get(`https://api.themoviedb.org/3/tv/${this.props.showID}/videos?api_key=${process.env.REACT_APP_TMDB_API_KEY}`).then(response => {
      if (response.data.results.length >= 1) {
        this.setState({
          youTubeKey: response.data.results[0].key
        })
      }
      console.log(response.data)
    })
  }



  render () {
    console.log(this.props, this.state)
    if (isNotNullOrUndefined(this.props.showID) || isNotNullOrUndefined(this.state.youTubeKey)) {
      return <></>
    } else {
      return (
        <div>
          <ModalVideo channel='youtube' isOpen={this.state.isOpen} videoId={this.state.youTubeKey} onClose={() => this.setState({isOpen: false})} />
          <Button onClick={this.openModal}>â–º View Trailer</Button>
        </div>
      )
    }
  }
}
