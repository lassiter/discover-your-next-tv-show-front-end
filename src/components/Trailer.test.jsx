import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components'
import toJson from 'enzyme-to-json'

import Trailer from './Trailer'

describe('App', function() {
  it('renders correctly with id value', function() {
    const wrap = shallow(
      <Trailer showID={1399}/>
    )
    wrap.setState({
      youTubeKey: "Dv0UBTfnqdI"
    })
    expect(toJson(wrap)).toMatchSnapshot()
  });

  it('renders correctly with id value', function() {
    const wrap = shallow(
      <Trailer showID={null}/>
    )
    expect(toJson(wrap)).toMatchSnapshot()
  });
});

