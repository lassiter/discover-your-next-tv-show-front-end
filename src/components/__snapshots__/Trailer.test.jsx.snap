// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly with id value 1`] = `
<div>
  <ModalVideo
    allowFullScreen={true}
    animationSpeed={300}
    aria={
      Object {
        "dismissBtnMessage": "Close the modal by clicking here",
        "openMessage": "You just openned the modal video",
      }
    }
    channel="youtube"
    classNames={
      Object {
        "modalVideo": "modal-video",
        "modalVideoBody": "modal-video-body",
        "modalVideoClose": "modal-video-close",
        "modalVideoCloseBtn": "modal-video-close-btn",
        "modalVideoEffect": "modal-video-effect",
        "modalVideoIframeWrap": "modal-video-movie-wrap",
        "modalVideoInner": "modal-video-inner",
      }
    }
    isOpen={false}
    onClose={[Function]}
    ratio="16:9"
    videoId="Dv0UBTfnqdI"
    vimeo={
      Object {
        "api": false,
        "autopause": true,
        "autoplay": true,
        "byline": true,
        "callback": null,
        "color": null,
        "height": null,
        "loop": false,
        "maxheight": null,
        "maxwidth": null,
        "player_id": null,
        "portrait": true,
        "title": true,
        "width": null,
        "xhtml": false,
      }
    }
    youku={
      Object {
        "autoplay": 1,
        "show_related": 0,
      }
    }
    youtube={
      Object {
        "autoplay": 1,
        "cc_load_policy": 1,
        "color": null,
        "controls": 1,
        "disablekb": 0,
        "enablejsapi": 0,
        "end": null,
        "fs": 1,
        "h1": null,
        "iv_load_policy": 1,
        "list": null,
        "listType": null,
        "loop": 0,
        "modestbranding": null,
        "origin": null,
        "playlist": null,
        "playsinline": null,
        "rel": 0,
        "showinfo": 1,
        "start": 0,
        "theme": "dark",
        "wmode": "transparent",
      }
    }
  />
  <styled.button
    onClick={[Function]}
  >
    ► View Trailer
  </styled.button>
</div>
`;

exports[`App renders correctly with id value 2`] = `<Fragment />`;

exports[`renders correctly with id value 1`] = `null`;

exports[`renders correctly with null value 1`] = `
<div>
  <button
    className="sc-bdVaJa eZJXyw"
    onClick={[Function]}
  >
    ► View Trailer
  </button>
</div>
`;
